app/controllers/departments_controller.rb

```diff
@@ -0,0 +1,57 @@
+class DepartmentsController < ApplicationController
+
+
+  def index
+    @departments = Department.sorted
+    render (:index)
+  end
+
+  def show
+    @department = Department.find(params[:id])
+  end
+
+  def new
+    @department = Department.new
+  end
+
+  def create
+    @department = Department.new(department_params)
+    if @department.save
+      flash[:notice] = "Department created successfully!"
+      redirect_to(departments_path)
+    else
+      render(:new)
+    end
+  end
+
+  def edit
+    @department = Department.find(params[:id])
+  end
+
+  def update
+    @department = Department.find(params[:id])
+    if @department.update_attributes(department_params)
+      flash[:notice] = "Department updated successfully!"
+      redirect_to(departments_path(@department))
+    else
+      render('edit')
```

Comment:
We can use symbol like 'render :edit` instead of `render('edit')`. you can get help from this [link](https://guides.rubyonrails.org/layouts_and_rendering.html)